src/main/java/de/medizininformatik_initiative/process/report/service/CheckSearchBundle.java

```diff
@@ -88,56 +92,84 @@ private void testNoResources(List<Bundle.BundleEntryComponent> searches)
 
 	private void testRequestMethod(List<Bundle.BundleEntryComponent> searches)
 	{
+		long searchesCount = searches.size();
 		long httpGetCount = searches.stream().filter(Bundle.BundleEntryComponent::hasRequest)
 				.map(Bundle.BundleEntryComponent::getRequest).filter(Bundle.BundleEntryRequestComponent::hasMethod)
 				.map(Bundle.BundleEntryRequestComponent::getMethod).filter(Bundle.HTTPVerb.GET::equals).count();
 
-		int searchesCount = searches.size();
-
 		if (searchesCount != httpGetCount)
 			throw new RuntimeException("Search Bundle contains HTTP method other then GET");
 	}
 
 	private void testRequestUrls(List<Bundle.BundleEntryComponent> searches)
 	{
+		int searchesCount = searches.size();
 		List<Bundle.BundleEntryRequestComponent> requests = searches.stream()
 				.filter(Bundle.BundleEntryComponent::hasRequest).map(Bundle.BundleEntryComponent::getRequest)
 				.filter(Bundle.BundleEntryRequestComponent::hasUrl).toList();
-
 		int requestCount = requests.size();
-		int searchesCount = searches.size();
 
 		if (searchesCount != requestCount)
 			throw new RuntimeException("Search Bundle contains request without url");
 
 		List<UriComponents> uriComponents = requests.stream()
-				.map(r -> UriComponentsBuilder.fromUriString(r.getUrl()).build()).collect(Collectors.toList());
+				.map(r -> UriComponentsBuilder.fromUriString(r.getUrl()).build()).toList();
 
-		testContainsSummaryCount(uriComponents);
+		testContainsNoPathIds(uriComponents);
+		testContainsValidSummaryCount(uriComponents);
 		testContainsValidSearchParams(uriComponents);
 		testContainsValidDateSearchParams(uriComponents);
-		testContainsValidCodeSearchParams(uriComponents);
+		testContainsValidTokenSearchParams(uriComponents);
+	}
+
+	private void testContainsNoPathIds(List<UriComponents> uriComponents)
+	{
+		// Resource type is only present in IdType if URI path contains a resource id
+		List<IdType> pathsWithId = uriComponents.stream().map(UriComponents::getPath).filter(Objects::nonNull)
```

Comment:
Would it be more concise to test that the Path is one of the enum constants from `org.hl7.fhir.r4.model.ResourceType`? Have you looked into the `setValue(String)` method of `IdType`? I would't be sure that I can't trick it to write a null `myResourceType`.

Reply:
implemented with 4f143c1